modules/EvseManager/EvseManager.cpp

```diff
@@ -1242,29 +1240,13 @@ bool EvseManager::get_hlc_waiting_for_auth_pnc() {
     return hlc_waiting_for_auth_pnc;
 }
 
-void EvseManager::log_v2g_message(Object m) {
-    std::string msg = m["v2g_message_id"];
-
-    std::string xml = "";
-    std::string json_str = "";
-    std::string exi_hex = "";
-    std::string exi_base64 = "";
+void EvseManager::log_v2g_message(types::iso15118_charger::V2gMessages v2g_messages) {
 
-    if (not m["v2g_message_xml"].is_null()) {
-        json_str = m["v2g_message_json"];
-    }
-
-    if (not m["v2g_message_json"].is_null()) {
-        xml = m["v2g_message_xml"];
-    }
-
-    if (not m["v2g_message_exi_hex"].is_null()) {
-        exi_hex = m["v2g_message_exi_hex"];
-    }
-
-    if (not m["v2g_message_exi_base64"].is_null()) {
-        exi_base64 = m["v2g_message_exi_base64"];
-    }
+    const std::string msg = types::iso15118_charger::v2g_message_id_to_string(v2g_messages.v2g_message_id);
+    const std::string xml = v2g_messages.v2g_message_xml.value_or("");
```

Comment:
The struct `V2gMessages` should probably have member properties called `xml` instead of `v2g_message_xml`, so it won't read that redundant like `v2g_messages.v2g_message_xml`.  Probably, `V2gMessages` might also be written as `V2GMessages`?
But in general it is good to have the information passed as a typed struct, instead of *untyped* json :+1:

Reply:
Yeah maybe it is time to refactor the `V2gMessages` type in types/iso15118_charger. With `PascalCase` I think it is `V2gMessages` and not `V2GMessages`

Reply:
Refactoring done